File: java/LingosHookAndroidClient2/src/jie/java/android/lingoshook/data/DBAccess.java
package jie.java.android.lingoshook.data;

import jie.java.android.lingoshook.Global;
import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteException;
import android.os.Bundle;
import android.util.Log;

public class DBAccess extends BaseDataObject {
	
	public static final DBAccess instance = new DBAccess();
	
	private static final String DATABASE_FILE			=	"/data/data/jie.java.android.lingoshook/LingosHook.db";;
	public static final int DATABASE_FORMAT_VERSION		=	2;
	
	private static final String TABLE_TEST			=	"Test";
	private static final String TABLE_WORD			=	"Word";
	private static final String TABLE_SRC_INDEX		=	"Src_Index";
	private static final String TABLE_SRC_TEXT_INDEX	=	"Src_Text_Index";
	private static final String TABLE_SRC_TEXT_MEANING	=	"Src_Text_Meaning";
	private static final String TABLE_SRC_HTML		=	"Src_Html";
	private static final String TABLE_DICTIONARY	=	"Dictionary";
	private static final String TABLE_SCORE			=	"Score";
	
	private static final String COLUMN_ID		=	"id";
	private static final String COLUMN_VALUE	=	"value";
		
	private static final String COL_WORDID		=	"wordid";
	private static final String COL_WORD		=	"word";
	private static final String COL_FLAG		=	"flag";
	private static final String COL_SRCID		=	"srcid";
	private static final String COL_FMT			=	"fmt";
	private static final String COL_ORIG		=	"orig";
	private static final String COL_DICTID		=	"dictid";
	private static final String COL_SYMBOL		=	"symbol";
	private static final String COL_CATEGORY	=	"category";
	private static final String COL_MEANING		=	"meaning";
	private static final String COL_HTML		=	"html";
	private static final String COL_TITLE		=	"title";
	private static final String COL_LAST		=	"last";
	private static final String COL_NEXT		=	"next";//next update
	private static final String COL_SCORE		=	"score";	
	
	private static SQLiteDatabase db = null;

	@Override
	public int init(Bundle savedInstanceState) {

		try {
			db = SQLiteDatabase.openOrCreateDatabase(DATABASE_FILE, null);
			
			Log.i(Global.APP_TITLE, "db initial - " + db.getMaximumSize());
		}
		catch (SQLiteException e) {
			return -1;
		}
		
		if(initTables() != 0)
			return -1;
		if(initData() != 0)
			return -1;
		
		return 0;		

	}
	
	@Override
	protected void release() {
		if(db != null) {
			db.close();
		}
	}
		
	protected int initTables() {
		
		try {
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_WORD + " (" 
						+ COL_WORDID + " INTEGER PRIMARY KEY," 
						+ COL_WORD + " TEXT,"
						+ COL_FLAG + " INTEGER)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SRC_INDEX + " ("
						+ COL_SRCID + " INTEGER PRIMARY KEY,"					
						+ COL_WORDID + " INTEGER,"
						+ COL_FMT + " INTEGER,"
						+ COL_ORIG + " INTEGER)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SRC_TEXT_INDEX + " ("
						+ COL_SRCID + " INTEGER,"
						+ COL_DICTID + " INTEGER,"
						+ COL_SYMBOL + " TEXT)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SRC_TEXT_MEANING + " ("
						+ COL_SRCID + " INTEGER,"
						+ COL_CATEGORY + " TEXT,"
						+ COL_MEANING + " TEXT)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SRC_HTML + " ("
						+ COL_SRCID + " INTEGER,"
						+ COL_HTML + " TEXT)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_DICTIONARY + " ("
						+ COL_DICTID + " INTEGER,"
						+ COL_TITLE + " TEXT)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS " + TABLE_SCORE + " ("
						+ COL_WORDID + " INTEGER,"
						+ COL_LAST + " INTEGER,"
						+ COL_NEXT + " INTEGER,"
						+ COL_SCORE + " SCORE)");
			
			db.execSQL("CREATE TABLE IF NOT EXISTS Test (id INTEGER PRIMARY KEY,value TEXT)");
		}
		catch (SQLiteException e) {
			return -1;
		}
		
		
		return 0;
	}
	
	protected int initData() {
		
		ContentValues values = new ContentValues();
		
		for(int i = 0; i < 24; ++ i) {
			values.clear();
			
			values.put(COLUMN_ID, String.valueOf(i));
			values.put(COLUMN_VALUE, String.format("%d - This is %d", i, i));
			db.insert(TABLE_TEST, null, values);
		}
		
		return 0;
	}

	public Cursor getTestData(String condition, int offset, int maxRows) {
		String sql = "SELECT id, value FROM Test";
		if(condition != null) {
			sql += " WHERE " + condition;
		}
		sql += " LIMIT " + maxRows + " OFFSET " + offset ;
		
		return db.rawQuery(sql, null);
	}
	
	
}

File: temp_proj/python/sgmllib_test/src/dbaccess.py
#!/usr/bin/python
# coding:utf-8

import sqlite3 as sqlite
import re


def table_create(conn):
    cursor = conn.cursor()
    
    sql = [
           'CREATE TABLE IF NOT EXISTS Word (wordid INTEGER SECOND KEY, word TEXT PRIMARY KEY, flag INTEGER)',
           'CREATE TABLE IF NOT EXISTS Src (srcid INTEGER PRIMARY KEY, wordid INTEGER, fmt INTEGER, orig INTEGER, content TEXT)',
           'CREATE TABLE IF NOT EXISTS Dict (dictid INTEGER PRIMARY KEY, title TEXT)',
           'CREATE TABLE IF NOT EXISTS Word2 (wordid INTEGER PRIMARY KEY, word TEXT, flag INTEGER, orig INTEGER, fmt INTEGER)',
           'CREATE TABLE IF NOT EXISTS ViconIndex (wordid INTEGER, position INTEGER, size INTEGER)'
           ]
    for s in sql:
        cursor.execute(s)

        
def add_dict(conn, title):
    cursor = conn.cursor()    
    cursor.execute('INSERT INTO Dict (title) VALUES (\'%s\')' % title)    
    conn.commit()        
        
def add_record(conn, cursor, word, record):
#    sql.encode('string_scape')
    cursor.execute('INSERT INTO Word (word, flag) VALUES ("%s",1)' % (word))#'INSERT INTO Word (word, flag) VALUES (\'%s\',1)' % (word))
    record = record.replace('\"', '')
    cursor.execute('INSERT INTO Src (wordid, fmt, orig, content) VALUES (%d, 3, 1, "%s")' % (cursor.lastrowid, record))
#    conn.commit()
    
def add_word(conn, word, pos, len):
    cursor = conn.cursor()
    cursor.execute('INSERT INTO Word2 (word, flag, orig, fmt) VALUES ("%s",1, 1, 4)' % (word))#'INSERT INTO Word (word, flag) VALUES (\'%s\',1)' % (word))
    cursor.execute('INSERT INTO ViconIndex (wordid, position, size) VALUES (%d, %d, %d)' % (cursor.lastrowid, pos, len))   
    
    
def db_create(dbfile):
    return sqlite.connect(dbfile)

def db_close(conn):
    conn.commit();
    conn.close()
    
def db_test(conn):
    cursor = conn.cursor()
    record = '"1234"'
    record.replace('\"', '')
    cursor.execute('INSERT INTO Word (word, flag) VALUES ("%s", 1)' % (record))
    conn.commit()
         

#######################################3

#conn = db_create('../data/lac.db')


#add_dict(conn, 'test')

#db_close(conn)
    

